packages/wdio-browserstack-service/src/scripts/test-event-scripts.ts

```diff
@@ -0,0 +1,78 @@
+export function testStartEvent() {
+    return `
+        const callback = arguments[arguments.length - 1]
+        const fn = () => {
+            window.addEventListener('A11Y_TAP_STARTED', fn2)
+            const e = new CustomEvent('A11Y_FORCE_START')
+            window.dispatchEvent(e)
+        }
+        const fn2 = () => {
+            window.removeEventListener('A11Y_TAP_STARTED', fn)
+            callback()
+        }
+        fn()
+    `
+}
+
+export function testForceStop() {
+    return `
+        const e = new CustomEvent('A11Y_FORCE_STOP')
+        window.dispatchEvent(e)
+    `
+}
+
+export function testStop() {
+    return `
+        const callback = arguments[arguments.length - 1];
+
+        this.res = null;
+        if (arguments[0].saveResults) {
+        window.addEventListener('A11Y_TAP_TRANSPORTER', (event) => {
+            window.tapTransporterData = event.detail;
+            this.res = window.tapTransporterData;
+            callback(this.res);
+        });
+        }
+        const e = new CustomEvent('A11Y_TEST_END', {detail: arguments[0]});
+        window.dispatchEvent(e);
+        if (arguments[0].saveResults !== true ) {
+        callback();
+        }
+    `
+}
+
+export function accessibilityResults() {
+    return `
+        return new Promise(function (resolve, reject) {
+            try {
+            const event = new CustomEvent('A11Y_TAP_GET_RESULTS');
+            const fn = function (event) {
+                window.removeEventListener('A11Y_RESULTS_RESPONSE', fn);
+                resolve(event.detail.data);
+            };
+            window.addEventListener('A11Y_RESULTS_RESPONSE', fn);
+            window.dispatchEvent(event);
+            } catch {
+            reject();
+            }
+        });
+    `
+}
+
+export function accessibilityResultsSummary() {
+    return `
```

Comment:
No need to return a string as it would remove all the code highlighting and prevents us from being able to test it. You can do this:

__scripts.ts__
```ts
export function accessibilityResultsSummary (paramA, paramB) {
  return new Promise(...
  // ...
}
```
in your service file:

```ts
import { accessibilityResultsSummary } from './scripts.js'

// ...
await browser.execute(accessibilityResultsSummary, paramA, paramB)
```

Reply:
> Unit Tests don't seem to be feasible to test scripts

How so? Vitest allows running in JSDOM mode and we could even test the function in the browser using WebdriverIO itself 😉